src/jit/codegenxarch.cpp

```diff
@@ -1813,162 +1813,20 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
                     regNumber retReg0 = REG_NA;
-                    emitAttr size0 = EA_UNKNOWN;
-                    unsigned offset0 = structDesc.eightByteOffsets[0];
+                    unsigned __int8 offset0 = 0;
```

Comment:
Both in Lowerxarch.cpp and Line 1801 above: the condition to detect struct return in two registers is

varTypeIsStruct(tree) && tree->gtGetOp1()->OperGet() == GT_LCL_VAR

Can there be a possibility that varTypeIsStruct(tree) is true but op1 is not GT_LCL_VAR?  Such an IR combination is not possible or how is it handled.  If such a combination is not possible, we should noway assert both in lowerxarch.cpp and codegenxarch.cpp rather than using it in condition.